optuna/pruners/successive_halving.py

```diff
@@ -0,0 +1,149 @@
+import math
+from typing import List  # NOQA
+
+from optuna.pruners.base import BasePruner
+from optuna.storages import BaseStorage  # NOQA
+from optuna.structs import FrozenTrial  # NOQA
+
+
+class SuccessiveHalvingPruner(BasePruner):
+
+    """Pruner using Asynchronous Successive Halving Algorithm.
+
+    `Successive Halving <https://arxiv.org/abs/1502.07943>`_ is a bandit-based algorithm to
+    identify the best one among multiple configurations. This class implements an asynchronous
+    version of Successive Halving. Please refer to the paper of
+    `Asynchronous Successive Halving <http://arxiv.org/abs/1810.05934>`_ for detailed descriptions.
+
+    Note that, this class does not take care of the parameter for the maximum
+    resource, referred to as ``R`` in the paper. The maximum resource allocated to a trial is
+    typically limited inside the objective function (e.g., ``step`` number in `simple.py
+    <https://github.com/pfnet/optuna/tree/c5777b3e/examples/pruning/simple.py#L31>`_,
+    ``EPOCH`` number in `chainer_integration.py
+    <https://github.com/pfnet/optuna/tree/c5777b3e/examples/pruning/chainer_integration.py#L65>`_).
+
+    Example:
+
+        We minimize an objective function with ``SuccessiveHalvingPruner``.
+
+        .. code::
+
+            >>> from optuna import create_study
+            >>> from optuna.pruners import SuccessiveHalvingPruner
+            >>>
+            >>> def objective(trial):
+            >>>     ...
+            >>>
+            >>> study = create_study(pruner=SuccessiveHalvingPruner())
+            >>> study.optimize(objective)
+
+    Args:
+        min_resource:
+            A parameter for specifying the minimum resource allocated to a trial
+            (in the `paper <http://arxiv.org/abs/1810.05934>`_ this parameter is
+            referred to as "r").
```

Comment:
Because back quotes are used to escape numerical expressions like `R` in L19, the quotation marks around `r` should be replaced with backquotes. Please take care of `eta` and `s` in addition to `r`.

Reply:
Thank you for pointing it out. I fixed this at the commit https://github.com/pfnet/optuna/pull/236/commits/429e27e392594ad3344d955f98d6adf69d341974.